index.bs

```diff
@@ -4332,6 +4344,22 @@ as seen for example in [[#example-ws-no-backpressure]].
  1. Return ! [$IsWritableStreamLocked$]([=this=]).
 </div>
 
+<div algorithm>
+ The <dfn id="ws-write" method for="WritableStream">write(|chunk|)</dfn> method steps are:
```

Comment:
Would we want to imply from the name that each write would be exclusive and extra parallel calls would reject?

Reply:
I just realized the behavior is is wrong. The writer should be released immediately, not once the write promise resolves. This mirrors the behavior of the write method on https://fs.spec.whatwg.org/#filesystemwritablefilestream

Reply:
That sounds better!

Reply:
Releasing the writer works so long as releasing it does not lead the write being canceled. I believe it's fine now but I believe a [change](https://streams.spec.whatwg.org/#abstract-opdef-readablestreamdefaultreadererrorreadrequests) was recently made to `ReadableStream` that caused pending reads to be canceled. Are we certain a similar type of change won't be wanted with writers and would this interfere with that?

Reply:
That's very unlikely to change.

For readers, we previously did not allow you to `releaseLock()` while there were pending read requests. For writers, we never had that restriction: you're always allowed to call `releaseLock()` even when there are pending write requests. In fact, [we even recommend](https://streams.spec.whatwg.org/#example-manual-write-dont-await) not awaiting the `writer.write()` promise so you can better respect backpressure (as long as you await `writer.ready` though).